### PR TITLE
oidc docs & some refactoring

### PR DESCRIPTION
**Pull request by jnummelin**

### CODE REVIEW

File: configuration/authentication.md [hunk 1]
```diff
@@ -0,0 +1,44 @@
+### `authentication`
```
Comment (jakolehm) [inline configuration/authentication.md]:
```suggestion
# Configuration: Authentication
```

File: configuration/authentication.md [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+Configure cluster authentication mechanisms. To understand how authentication and authorization works in Kubernetes clusters, see https://kubernetes.io/docs/reference/access-authn-authz/authentication/
+
+#### `token_webhook`
```
Comment (jakolehm) [inline configuration/authentication.md]:
```suggestion
## `token_webhook`
```

File: configuration/authentication.md [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+        server: http://localhost:9292/token
+        certificate_authority: /path/to/ca.pem # optional
+      user:
+        name: kube-apiserver
+        client_key: /path/to/key.pem # optional
+        client_certificate: /path/to/cert.pem # optional
+    cache_ttl: 5m # optional
+```
+
+#### `oicd`
```
Comment (jakolehm) [inline configuration/authentication.md]:
```suggestion
## `oicd`
```

File: configuration/README.md [hunk 4]
```diff
@@ -234,25 +234,6 @@ The supported configuration options:
 
 Audit events are delivered in batched mode, multiple events in one webhook `POST` request.
 
-### `authentication`
```
Comment (jakolehm) [inline configuration/README.md]: Maybe leave this here as with `network`?

Comment (jnummelin) [reply configuration/README.md]: done

**Comment by jakolehm** [comment]
Conflicts...

**Comment by jnummelin** [comment]
rebased

**Pull request merged**